server/src/routes/transaction_controller.rs

```
@@ -0,0 +1,225 @@
+	#[display(fmt = "Mutation has to be negative for this transaction type")]
+	BadNegativeMutation
+}
+
+impl error::ResponseError for TransactionError {
+	fn error_response(&self) -> HttpResponse<actix_web::body::BoxBody> {
+			HttpResponse::build(self.status_code())
+				.json(APIError { error: self.to_string() })
+	}
+
+	fn status_code(&self) -> StatusCode {
+		match *self {
+			Self::NoTransactions => StatusCode::NOT_FOUND,
+			Self::BadTransactionId => StatusCode::BAD_REQUEST,
+			Self::InternalServerError => StatusCode::INTERNAL_SERVER_ERROR,
+			Self::BadMutation => StatusCode::BAD_REQUEST,
+			Self::BadPositiveMutation => StatusCode::BAD_REQUEST,
+			Self::BadNegativeMutation => StatusCode::BAD_REQUEST
+		}
+	}
+}
+
+//Getting transactions
+
+#[get("/user/{userId}/transactions")]
+pub async fn get_user_transactions(path: web::Path<String>,  pool: web::Data<DbPool>) -> Result<HttpResponse, TransactionError> {
+	let user_id = path.into_inner();
+
+	let user_transactions: Vec<Transaction> = web::block(move || {
+		use crate::schema::transactions::dsl::*;
+		let mut connection = pool.get()
+			.expect("Failed to get connection from pool");
+
+		transactions
+			.filter(account.eq(user_id))
+			.load::<Transaction>(&mut connection)
+			.expect("Error fetching transactions")
+	})
+	.await
+	.map_err(|_e| TransactionError::NoTransactions)?;
+
+	if user_transactions.is_empty() {
+		return Err(TransactionError::NoTransactions);
+	}
+
+	Ok(HttpResponse::Ok().json(user_transactions))
+}
+
+#[get("/transactions/{transactionId}")]
+pub async fn get_transaction(path: web::Path<String>,  pool: web::Data<DbPool>) -> Result<HttpResponse, TransactionError> {
+	let transaction_id = path.into_inner().parse::<i32>().map_err(|_e| TransactionError::BadTransactionId)?;
+
+	let user_transactions: Vec<Transaction> = web::block(move || {
+		use crate::schema::transactions::dsl::*;
+		let mut connection = pool.get()
+			.map_err(|_e| TransactionError::NoTransactions)
```

Are we swallowing this errors in the `map_err` below?

---

Just noticed this in the TODO below, we can chuck a TODO here too to remind future us

---

Any idea how we could handle this? Maybe a match case to map them to our own, cool errors?

---

Also, should we just make one giant enum for errors and how they're serialized? Or could we write something extendable?

---

I think the lower todo is enough, don't want a future edit to just remove one.